stm32l4-multi/tty: add RX fifo buffer written in IRQ

xvuko opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
16 byte FIFO is added between RX interrupt and tty processing thread.

## Motivation and Context
Without this change TTY driver occasionally looses RX bytes. 

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

<!--- In case of breaking change - please advice here what needs to be done in dependent projects. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: stm32l4

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

agkaminski commented on multi/stm32l4-multi/tty.c:
```diff
@@ -115,19 +118,28 @@ static void tty_irqthread(void *arg)
+		ctx->rxready = 0;
+		dataBarier();
+
+		for (rxcount = 8; rxcount && lf_fifo_pop(&ctx->rx_fifo, &rxbyte); rxcount--) {
```
Why 8? Does sizeof(rx_fifo_buffer) / 2 = rxcount is true?

agkaminski reviewed (commented):
I don't really like two things:
- fifo code is copied, it's just a header, sharing it should be easy,
- fifo code uses compiler extentions which is a MISRA violation.

xvuko commented:
>     * fifo code is copied, it's just a header, sharing it should be easy,
I moved lf_fifo to libtty as public header.  Added TODO about moving into corelibs.
